lib/fine-types/app/Options.hs

```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE ApplicativeDo #-}
+{-# LANGUAGE LambdaCase #-}
+
+module Options where
+
+import Prelude
+
+import Options.Applicative
+    ( ReadM
+    , command
+    , eitherReader
+    , execParser
+    , fullDesc
+    , header
+    , help
+    , helper
+    , hsubparser
+    , info
+    , long
+    , metavar
+    , option
+    , progDesc
+    , short
+    , strArgument
+    , (<**>)
+    )
+
+newtype Options = Options
+    { optCommand :: Commands
+    }
+
+newtype Commands
+    = Convert ConvertOptions
+
+data Format = Json | Yaml
+
+data ConvertOptions = ConvertOptions
+    { optInput :: FilePath
+    , optOutput :: FilePath
+    , optFormat :: Format
+    }
+
+parseOptions :: IO Options
+parseOptions =
+    execParser
+        $ info (options <**> helper)
+        $ fullDesc
+            <> progDesc "Run fine-types commands"
+            <> header "fine-types - a tool for working with fine-types modules"
+  where
+    options = Options <$> commands
+
+    commands =
+        hsubparser
+            $ mconcat
+                [ command "convert" $ info (Convert <$> convertOptions) mempty
+                -- other commands go here
+                ]
+
+    convertOptions = do
+        input <- strArgument $ metavar "FILE" <> help "input file"
+        output <- strArgument $ metavar "FILE" <> help "output file"
```

Comment:
It's a bit bike shedding, but I believe that the tool would be slightly less error-prone to use if one (or even both) of the file arguments are behind a flag, e.g. `-o` for the output file. 🤔 

Having to specify `-o` is the convention for compilers like GHC.
The convention to specify `-i` is used for [`ffmpeg`](https://ffmpeg.org/ffmpeg.html#Synopsis). 🙈

Reply:
my first version was like that